### DISCUSSION TITLE
Can PHPCBF autofix line length errors?

### DISCUSSION THREAD

**Discussion by puka-tchou** [squizlabs/PHP_CodeSniffer#3221] [Q&A] [answered]
Hey, I'm progressively integrating PHPCS as a part of our quality testing and we are getting a lot of line length errors. Can PHPCBF fix these issues and if not is this doable?

This would really help us to correct our code much faster.

**Comment by gsherwood** [comment, marked as answer]
No, PHPCBF cannot auto-fix line length as it doesn't have rules for where to break a statement up in a way that makes sense to a developer. I don't think this is something that PHPCBF will ever do as it's really something that a developer should take a look at and see if there are other ways to shorten the line. Simply cutting it at the break-point isn't really the point of the line length sniff.

If you don't want line length checked, you can [create a custom ruleset.xml file](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Annotated-Ruleset) and just remove that check.

**Reply by puka-tchou** [reply]
Thanks for the answer! Would you have any idea of another tool that could do so?

**Reply by gsherwood** [reply]
I don't know of one, no.

**Reply by simonLeary42** [reply]
[the official Prettier PHP Plugin](https://github.com/prettier/plugin-php) can autofix line length, but it doesn't strictly adhere to PSR-12.
